python/pyproject.toml

```diff
@@ -26,13 +35,15 @@ classifiers = [
 [tool.poetry.dependencies]
 python = ">=3.9 <3.13"
 nose = "1.3.7"
-pyspark = "^3.4"
+pyspark = ">=3.3"
```

Comment:
@SemyonSinchenko this means that Spark 4 will 'just work'?

Reply:
I don't think so, tbh. There are breaking changes, especially in the connect. I would like to have a separate issue to track 4.0-compatibility. Let me fix it.

Reply:
Fixed.